### DISCUSSION TITLE
Some problems I encountered when using the controlnet API

### DISCUSSION THREAD

**Discussion by wym19970515** [Mikubill/sd-webui-controlnet#270] [Q&A]
Problem 1:
    The default json parameter setting will cause some errors (only some errors, but the interface can still work normally in fact)：
    ![image](https://user-images.githubusercontent.com/45259608/220452868-0fc598c0-61d3-4af4-8a98-8bb8ea518ae6.png)
    ![image](https://user-images.githubusercontent.com/45259608/220453302-742a2d13-965d-4325-84e1-fb54f85d342d.png)
    When I solve the problem of hr_upscale type error, there will be new error reports, which will not be listed here.

Problem 2 (serious):
    The call of controlnet txt2img API (/controlnet/txt2img) will affect the original API (/sdapi/v1/txt2img and /sdapi/v1/img2img).
    As long as the controlnet API has been used, the use of the original API will be abnormal, and the line draft which is generated by controlnet before will still take effect on the call of the original API. Even after calling the controlnet API, the txt2img and img2img of the web page will still be affected by the line draft generated by the previous controlnet API, even if the controlnet function is not enabled.

By the way, the call of the controlnet api will generate a large number of debug logs, so I want to know whether the log level of the interface can be adjusted ?
![image](https://user-images.githubusercontent.com/45259608/220458783-7536ebeb-af1c-4176-b87e-6edf3abbca11.png)

**Comment by wym19970515** [comment]
Picture for Problem 2:
The line draft I obtained through controlnet txt2img API (/controlnet/txt2img)：
![398273613-2686355968-016DA01F374321601A3E1ECFE89AAEF8](https://user-images.githubusercontent.com/45259608/220460262-347c85b9-029a-4040-9c15-c437ec8d699d.png)
After the controlnet API call, the image generated through web page when controlnet is not enabled：
![image](https://user-images.githubusercontent.com/45259608/220460548-0d93e101-4ce0-4732-82ff-dc4816bdac9d.png)
Images generated many times can always fit with the line draft, and the time consumption is close to that of opening the controlnet function, so I think the online draft obtained through the control API is still valid at this time.

**Reply by Mikubill** [reply]
API args issue and network restore already fixed in recent commits.

**Reply by wym19970515** [reply]
Thanks
